app/code/Magento/Backend/view/adminhtml/templates/store/switcher.phtml

```diff
@@ -8,222 +8,132 @@
-            var val = $(event.target).data('value');
-            var role = $(event.target).data('role');
-            var switcher = $('[data-role='+role+']');
-
-            event.preventDefault();
-
-            if (!switcher.val() || val != switcher.val()) {
-                switcher.val(val).trigger('change'); // Set the value & trigger event
-            }
-        });
-    })(jQuery);
-
-    var scopeSwitcherHandler;
-
-    function switchScope(obj) {
-        var switcher = jQuery(obj);
-        var scopeId = switcher.val();
-        var scopeParams = '';
-        if (scopeId) {
-            scopeParams = switcher.data('param') + '/' + scopeId + '/';
-        }
-
-        if (obj.switchParams) {
-            scopeParams += obj.switchParams;
-        }
-
-        if ((typeof scopeSwitcherHandler) != 'undefined') {
-            var switcherParams = {
-                scopeId: scopeId,
-                scopeParams: scopeParams,
-                useConfirm: {$useConfirm}
-            };
-            scopeSwitcherHandler(switcherParams);
-        } else {
-script;
-    if ($block->getUseConfirm()) {
-                $scriptString .= '
-            confirm({
-                content:  "' . $block->escapeJs(__(
-                    'Please confirm scope switching. All data that hasn\'t been saved will be lost.'
-                )) . '",
-                actions: {
-                    confirm: function() {
-                        reload();
-                    },
-                    cancel: function() {
-                        obj.value = \'' . $block->escapeJs($block->getStoreId()) . '\';
-                    }
-                }
-            });
-';
-    } else {
-        $scriptString .= 'reload();';
-    }
-    $scriptString .= '
-        }
 
-        function reload() {
-        ';
-    if (!$block->isUsingIframe()) {
-        $scriptString .= '
-                var url = \'' . $block->escapeJs($block->getSwitchUrl()) . '\' + scopeParams;
-                setLocation(url);
-';
-    } else {
-        $scriptString .= <<<'script'
-            jQuery('#preview_selected_store').val(scopeId);
-            jQuery('#preview_form').submit();
-
-            jQuery('.store-switcher .dropdown-menu li a').each(function() {
-                var $this = jQuery(this);
-
-                if ($this.data('role') === 'store-view-id' && $this.data('value') == scopeId) {
-                    jQuery('#store-change-button').html($this.text());
+    <script type="text/x-magento-init">
+        {
+            "#store-switcher-backend": {
+                "Magento_Backend/js/store-switcher": {
+                    "useConfirm": <?= /* @noEscape */ $block->getUseConfirm(); ?>,
+                    "isUsingIframe": <?= /* @noEscape */ $block->isUsingIframe(); ?>,
+                    "switchUrl": "<?= $block->escapeUrl($block->getSwitchUrl()); ?>",
+                    "storeId": <?= /* @noEscape */ $block->getStoreId(); ?>
```

Comment:
I bevieve we should convert it to int before printing

Reply:
modified as your suggestion